prometheus_client/openmetrics/exposition.py

```diff
@@ -67,24 +67,87 @@ def generate_latest(registry):
                             floatToGoString(s.exemplar.value),
                         )
                 else:
-                    exemplarstr = ''
+                    exemplarstr = ''   
+                        
                 timestamp = ''
                 if s.timestamp is not None:
                     timestamp = f' {s.timestamp}'
+                
+                native_histogram = ''
+                positive_spans = ''
+                positive_deltas = ''
+                negative_spans = ''
+                negative_deltas = ''
+                pos = False
+                neg = False
+                     
+                if s.native_histogram:                
+                    # Initialize basic nh template
+                    nh_sample_template = '{{count:{},sum:{},schema:{},zero_threshold:{},zero_count:{}'
+
+                    args = [
+                        s.native_histogram.count_value,
+                        s.native_histogram.sum_value,
+                        s.native_histogram.schema,
+                        s.native_histogram.zero_threshold,
+                        s.native_histogram.zero_count,
+                    ]
+
+                    # Signal presence for pos/neg spans/deltas
+                    pos = False
+                    neg = False
+
+                    # If there are pos spans, append them to the template and args
+                    if s.native_histogram.pos_spans:
+                        positive_spans = ','.join([f'{ps[0]}:{ps[1]}' for ps in s.native_histogram.pos_spans])
+                        positive_deltas = ','.join(f'{pd}' for pd in s.native_histogram.pos_deltas)
+                        nh_sample_template += ',positive_spans:[{}]'
+                        args.append(positive_spans)
+                        pos = True
+
+                    # If there are neg spans exist, append them to the template and args
+                    if s.native_histogram.neg_spans:
+                        negative_spans = ','.join([f'{ns[0]}:{ns[1]}' for ns in s.native_histogram.neg_spans])
+                        negative_deltas = ','.join(str(nd) for nd in s.native_histogram.neg_deltas)
+                        nh_sample_template += ',negative_spans:[{}]'
+                        args.append(negative_spans)
+                        neg = True
+
+                    # Append pos deltas if pos spans were added
+                    if pos:
+                        nh_sample_template += ',positive_deltas:[{}]'
+                        args.append(positive_deltas)
+
+                    # Append neg deltas if neg spans were added
+                    if neg:
+                        nh_sample_template += ',negative_deltas:[{}]'
+                        args.append(negative_deltas)
```

Comment:
Couldn't these just be included in the already present if statements above and then there would be no need for the `pos` and `neg` variables?

Reply:
In order to do that, assuming we don't want to print out the keys of unpopulated fields, I would have to compose all the partial strings before knowing if they'll have to be added or not and then write the necessary conditionals along the lines of:

If pos spans and not neg spans:
   add pos spans and pos deltas

If pos spans and neg spans:
   add pos spans neg spans pos deltas and neg deltas

If neg spans and not pos spans:
   add neg spans and neg deltas

Maybe there are nicer solutions, but maybe now that I'm not shadowing `s.native_histogram.pos_spans` and `s.native_histogram.neg_spans` anymore, it looks more acceptable than before? XD